## Rubric

Instructions: DELETE this cell before you submit via a `git push` to your repo before deadline. This cell is for your reference only and is not needed in your report. 

 Scoring: Out of 10 points

- Each Developing  => -2 pts
- Each Unsatisfactory/Missing => -4 pts
  - until the score is 0

If students address the detailed feedback in a future checkpoint they will earn these points back






|                                  | **Unsatisfactory**                                                                                                                                                                                                                                                                                                                        | **Developing**                                                                                                                                                                                                       | **Proficient**                                                                                                                                                                                            | **Excellent**                                                                                                                                                                            |
|----------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| **EDA relevance**                | EDA is mostly neither relevant to the question nor helpful in figuring out how to address the question. Or the EDA does address the question, but many obviously relevant variables / analyses / figures were not included. | EDA is partly irrelevant/unhelpful. EDA missed one or two obvioulsy relevant analysis (distributions of single variables or relationships between variables) | EDA includes the obviously relevant / helpful variables in addressing the question.                                                              | Thorough EDA fully explored the dataset                                                                                                                 |
| **EDA analysis and description** | Many of the analyses are poor choices (e.g., using means instead of medians for obviously skewed data), or are poorly described in the text, or do not aid understanding the data                                                                                                                                                     | Some of the analyses are poor choices, or are poorly described in the text, or do not aid understanding the data                                                                                                 | All analyses are correct choices. Only one or two have minor issues in the text descriptions supporting them. Mostly they fit well with other elements of the EDA and support understanding the data  | All analyses are correct choices with clear text descriptions supporting them. The figures fit well with the other elements of the EDA, producing a clear understanding of the data. |
| **EDA figures**                  | Many of the figures are poor plot choices (e.g., using a bar plot to represent a time series where it would be better to use a line plot) or have poor aesthetics (including colormap, data point shape/color, axis labels, titles, annotations, text legibility) or do not aid understanding the data                                | Some of the figures are poor plot choices or have poor aesthetics. Some figures do not aid understanding the data                                                                                                | All figures are correct plot choices. Only one or two have minor questionable aesthetic choices. The figures mostly fit well with the other elements of the EDA and support understanding the data    | All figures are correct plot choices with beautiful aesthetics. The figures fit well with the other elements of the EDA, producing a clear understanding of the data.                |





# COGS 108 - EDA Checkpoint

Instructions: REPLACE the contents of this cell with your team list and their contributions. Note that this will change over the course of the checkpoints

This is a modified [CrediT taxonomy of contributions](https://credit.niso.org). For each group member please list how they contributed to this project using these terms:
> Analysis, Background research, Conceptualization, Data curation, Experimental investigation, Methodology, Project administration, Software, Visualization, Writing – original draft, Writing – review & editing

Example team list and credits:
- Jaden Wu: Conceptualization
- Trenton Pham: Database
- Juhoan Cha
- Minh Tue Nguyen
- Shawn Le

# Research Question

To what extent do pre-college academic indicators predict first-year college academic performance? Using a dataset containing high school GPA, standardized test scores (SAT math and verbal), coursework rigor (e.g., number of AP/IB courses), and high school context (e.g., school competitiveness), we model students’ first-year college performance. This project evaluates the relative and combined predictive power of these variables using regression and classification approaches. We further analyze whether certain preparation profiles systematically overperform or underperform relative to model expectations. This predictive modeling task aims to identify which aspects of high school success transfer most effectively to early college performance.


## Background and Prior Work

Recent national discussions around college readiness have been brought into sharp focus at UC San Diego, where reports of a growing remedial math placement rate among incoming students have gained mainstream attention. As UCSD students, we were motivated by this issue after the Academic Senate released a Senate–Administration Working Group (SAWG) report analyzing admissions practices and student academic preparedness. The report highlights a notable increase in the number of incoming students requiring remedial-level math instruction, raising concerns about the alignment between pre-college preparation and college-level expectations.<a name="cite_ref-1"></a><sup>1</sup>
 This development has prompted broader questions about how well traditional indicators of academic readiness translate into early college success.

The SAWG report attributes part of this trend to changes in UCSD’s admissions landscape, including increased enrollment of students from under-resourced and lower-funded high schools.<a name="cite_ref-1"></a><sup>1</sup>
 While the report does not claim that students from these schools are incapable of succeeding, it emphasizes that disparities in access to advanced coursework and academic resources may affect readiness for college-level mathematics. Importantly, the report suggests that commonly used admissions metrics may not fully capture differences in academic preparation, motivating further investigation into which pre-college factors meaningfully predict early college performance.

Prior research and exploratory projects have examined similar questions using simplified academic datasets. One such example is a publicly documented Kaggle project that analyzed the relationship between high school GPA, SAT scores, and first-year college GPA using linear regression.<a name="cite_ref-2"></a><sup>2</sup>
 That project found that high school GPA exhibited a slightly stronger correlation with first-year college GPA than SAT scores, suggesting that sustained academic performance may be a better indicator of early college success than standardized testing alone. The analysis also explored gender differences, observing small performance disparities despite similar SAT and high school GPA distributions.

These findings align with broader education research suggesting that no single pre-college metric fully explains college outcomes. Geiser and Santelices (2007), using University of California admissions and outcomes data, report that high school GPA is a stronger and more consistent predictor of college success than standardized test scores across multiple outcomes.<a name="cite_ref-3"></a><sup>3</sup>
 In a community college placement context, Belfield and Crosta (2012) compared placement-test-only approaches with transcript-based predictors and found that using high school performance indicators can improve placement decisions and reduce misclassification.<a name="cite_ref-4"></a><sup>4</sup>
 More recent work by Allensworth and Clark (2020), which uses hierarchical modeling across high schools, shows that GPA remains a relatively stable predictor while test-score predictive power varies more across school contexts.<a name="cite_ref-5"></a><sup>5</sup>
 Together, these methods-based findings suggest that model performance depends on both variable choice and institutional context, not just the algorithm itself.

Building on both institutional reporting and prior data-driven projects, our work seeks to extend this line of inquiry by examining a broader set of pre-college academic indicators and their relationship to early college performance. In addition to SAT and high school GPA, prior studies motivate including variables tied to coursework rigor and educational context (for example, AP/IB exposure, first-generation status, and school-resource context) because these factors may mediate how test scores translate into first-year outcomes.<a name="cite_ref-4"></a><sup>4</sup><a name="cite_ref-5"></a><sup>,5</sup>
 By situating our analysis within the context of the UCSD remedial math discussion and grounding our variable selection in existing methods-focused literature, this project aims to better understand the practical limits of commonly used readiness metrics and identify which aspects of high school preparation are most strongly associated with first-year college success.

References

<a name="cite_note-1"></a> 1
 UC San Diego Academic Senate. Report of the Senate–Administration Working Group on Admissions Review. UCSD, 2023. https://senate.ucsd.edu/media/740347/sawg-report-on-admissions-review-docs.pdf

<a name="cite_note-2"></a> 2
 Kaggle Discussion. Predicting College Success Using SAT and GPA. Kaggle, 2024. https://www.kaggle.com/discussions/getting-started/542760#3036652

<a name="cite_note-3"></a> 3
 Geiser, S., & Santelices, M. V. (2007). Validity of High-School Grades in Predicting Student Success Beyond the Freshman Year. UC Berkeley Center for Studies in Higher Education. https://cshe.berkeley.edu/publications/validity-high-school-grades-predicting-student-success-beyond-freshman-year-high-school

<a name="cite_note-4"></a> 4
 Belfield, C. R., & Crosta, P. M. (2012). Predicting Success in College: The Importance of Placement Tests and High School Transcripts. Community College Research Center, Teachers College, Columbia University. https://ccrc.tc.columbia.edu/publications/predicting-success-placement-tests-transcripts.html

<a name="cite_note-5"></a> 5
 Allensworth, E., & Clark, K. (2020). High school GPAs and ACT scores as predictors of college completion: Examining assumptions about consistency across high schools. Educational Researcher. https://doi.org/10.3102/0013189X20902110

# Hypothesis


Our group hypothesises that while high school GPA and SAT scores are a stronger indicator for college success through metrics such as gpa, environmental context factors (e.g., school structure and socioeconomic context) also play a gigantic role beyond academics alone.  Students that are set up well in good envionments alongside taking rigorous classes, have a much better chance of succeeding within college, whether its having higher first-year GPA, more credits earned, or a lower remedial-course rates

## Data

1) 
- The ideal dataset for this project would include a range of academic and demographic variables carefully selected based on their established relationship with first-year college GPA in existing academic literature. The primary predictor variables would include SAT Score (Math and Verbal), High School GPA, number of AP/IB courses completed, demographic background, gender, and first-generation college student status; commonly cited factors towards college performance. A sample size of at least 5,000 entries is targeted to ensure the model has enough data to identify meaningful patterns, reduce variance in predictions, and produce results that generalize well beyond the training set. A smaller sample risks producing an underfitted or unreliable model, particularly when accounting for the diversity of demographic subgroups within the data.

2) 
- One dataset have currently been identified on https://nces.ed.gov/surveys/hsls09/hsls09_data.asp, a platform that provides freely accessible statistical datasets from a high school to college/post-secondary education longtitudinal study. The data was collected from 2009-2016 that tracks students from secondary education to college, and each contains variables relevant to this project including first-year college GPA, cumulative high school GPA, SAT Math and SAT Verbal scores, first-generation status, demographic background, and gender. This dataset contains over 23,000 entries, way beyond our ideal sample size.

- A key consideration going forward is how the dataset will be used in the modeling process. We are currently extracting features to be used for the modeling process and an outcome variable. A final decision will be made after a thorough review of each dataset's structure, variable definitions, and any accompanying documentation.

In [1]:
# Run this code every time when you're actively developing modules in .py files.  It's not needed if you aren't making modules
#
## this code is necessary for making sure that any modules we load are updated here 
## when their source code .py files are modified

%load_ext autoreload
%autoreload 2

In [2]:
# Setup code -- Run only once after cloning!!! 
#
# this code downloads the data from its source to the `data/00-raw/` directory
# if the data hasn't updated you don't need to do this again!

# if you don't already have these packages (you should!) uncomment this line
# %pip install requests tqdm

import sys
sys.path.append('./modules') # this tells python where to look for modules to import

import get_data # this is where we get the function we need to download data

# replace the urls and filenames in this list with your actual datafiles
# yes you can use Google drive share links or whatever
# format is a list of dictionaries; 
# each dict has keys of 
#   'url' where the resource is located
#   'filename' for the local filename where it will be stored 
datafiles = [
    { 'url': 'https://raw.githubusercontent.com/fivethirtyeight/data/refs/heads/master/airline-safety/airline-safety.csv', 'filename':'airline-safety.csv'},
    { 'url': 'https://raw.githubusercontent.com/fivethirtyeight/data/refs/heads/master/bad-drivers/bad-drivers.csv', 'filename':'bad-drivers.csv'}
]

get_data.get_raw(datafiles,destination_directory='data/00-raw/')

Overall Download Progress:  50%|█████     | 1/2 [00:00<00:00,  4.41it/s]

Successfully downloaded: airline-safety.csv


Overall Download Progress: 100%|██████████| 2/2 [00:00<00:00,  4.53it/s]

Successfully downloaded: bad-drivers.csv


### Dataset #1 

### Secondary Education to College to  2009-2016 Longitudinal Study Data

Instructions: 
1. Change the header from Dataset #1 to something more descriptive of the dataset
2. Write a few paragraphs about this dataset. Make sure to cover
   1. Describe the important metrics, what units they are in, and giv some sense of what they mean.  For example "Fasting blood glucose in units of mg glucose per deciliter of blood.  Normal values for healthy individuals range from 70 to 100 mg/dL.  Values 100-125 are prediabetic and values >125mg/dL indicate diabetes. Values <70 indicate hypoglycemia. Fasting idicates the patient hasn't eaten in the last 8 hours.  If blood glucose is >250 or <50 at any time (regardless of the time of last meal) the patient's life may be in immediate danger"
   2. If there are any major concerns with the dataset, describe them. For example "Dataset is composed of people who are serious enough about eating healthy that they voluntarily downloaded an app dedicated to tracking their eating patterns. This sample is likely biased because of that self-selection. These people own smartphones and may be healthier and may have more disposable income than the average person.  Those who voluntarily log conscientiously and for long amounts of time are also likely even more interested in health than those who download the app and only log a bit before getting tired of it"
3. Use the cell below to 
    1. load the dataset 
    2. make the dataset tidy or demonstrate that it was already tidy
    3. demonstrate the size of the dataset
    4. find out how much data is missing, where its missing, and if its missing at random or seems to have any systematic relationships in its missingness
    5. find and flag any outliers or suspicious entries
    6. clean the data or demonstrate that it was already clean.  You may choose how to deal with missingness (dropna of fillna... how='any' or 'all') and you should justify your choice in some way
    7. You will load raw data from `data/00-raw/`, you will (optionally) write intermediate stages of your work to `data/01-interim` and you will write the final fully wrangled version of your data to `data/02-processed`
4. Optionally you can also show some summary statistics for variables that you think are important to the project
5. Feel free to add more cells here if that's helpful for you


# Csv file is too large to be pushed into git, so download it here
https://drive.google.com/file/d/1avtIqehIbvZ_4D7bibednlwA4rfM24GT/view?usp=sharing  


Raw file is super big, 20000 students with over 9000 columns.   
HSLS variable prefixes and what to use:

S = student survey items (raw questionnaire responses).   
X = derived/composite variables (cleaner model-ready summaries).   
P = parent survey items.   
W = survey weights (for population-representative estimates).   
## Wave mapping:

Base Year (2009): S1, X1, P1, W1.   
First Follow-up (2012): S2, X2, P2, W2.   
2013 Update + HS Transcript (2013-2014): S3, X3, W3.   
Second Follow-up (2016): S4, X4, W4.   
Postsecondary records/transcripts (2017-18): X5, W5 (no S5).   



In [6]:
import pandas as pd
from pathlib import Path

csv_path = Path("data/00-raw/hsls_17_student_pets_sr_v1_0.csv")
assert csv_path.exists(), f"Missing file: {csv_path.resolve()}"

# 1) Load raw data
df_raw = pd.read_csv(csv_path, low_memory=False)
print("Raw shape:", df_raw.shape)

# 2) Drop variables documented as Restricted-Use-File (RUF) only

df = df_raw.copy()

# 3) Drop columns that are entirely sentinel -5
all_neg5_cols = []
for col in df.columns:
    col_as_str = df[col].astype(str).str.strip()
    non_na = col_as_str[col_as_str.ne("nan")]
    if len(non_na) > 0 and non_na.eq("-5").all():
        all_neg5_cols.append(col)
df = df.drop(columns=all_neg5_cols)

# 4) Drop loan-related columns
loan_cols = [c for c in df.columns if "loan" in c.lower()]
df = df.drop(columns=loan_cols)

# 5) Treat remaining -5 as missing
df = df.replace(-5, pd.NA).replace("-5", pd.NA)

# 6) Drop low-information columns
constant_cols = [c for c in df.columns if df[c].nunique(dropna=True) <= 1]
df = df.drop(columns=constant_cols)

# 7) Drop high-missing columns
missing_threshold = 0.80
missing_ratio = df.isna().mean()
high_missing_cols = missing_ratio[missing_ratio >= missing_threshold].index.tolist()
df = df.drop(columns=high_missing_cols)

# 8) focused set of variables that we are interested in
selected_vars = [
    # 1) Demographic controls (objective)
    "X1SEX", "X1RACE", "X1HISPANIC", "X1WHITE", "X1BLACK",

    # 2) Family socioeconomic context (objective)
    "X1PAREDU", "X1FAMINCOME", "X1SES", "X1POVERTY130", "X1HHNUMBER",
    "X2PAREDU", "X2FAMINCOME", "X2SES", "X2POVERTY130", "X2HHNUMBER",

    # 3) School structural context (objective, not perceptions)
    "X1CONTROL", "X1LOCALE", "X1REGION", "X1SCHOOLCLI",
    "X2ENROLSTAT", "X2ENRSTATSCH", "X2NUMHS", "X2EVERDROP", "X2DROPSTAT",

    # 4) Academic achievement before college (objective tests/transcripts)
    "X1TXMTSCOR", "X1TXMQUINT", "X2TXMTSCOR", "X2TXMQUINT",
    "X3TGPATOT", "X3TGPAWGT", "X3TGPAACAD", "X3TGPASTEM",

    # 5) Coursework rigor and preparation (objective transcript indicators)
    "X3TNEWBASIC", "X3TCREDTOT", "X3TCREDACAD", "X3TCREDSTEM",
    "X3TCREDAPIB", "X3TCREDADV", "X3TCREDHON",
    "X3TCREDMAT", "X3TCREDSCI", "X3TCREDENG",
    "X3THIMATH", "X3THISCI",
    "X3T1CREDALG2", "X3T1CREDCALC", "X3T1CREDPHYS", "X3T1CREDSTAT",
    "X3TWHENALG1", "X3HSCOMPSTAT"
]


outcome_vars = [
    # First-year college success outcomes (postsecondary records)
    "X5YR1GPA", "X5YR1ATT", "X5YR1ERN", "X5REMRAT", "X5ENRATT"
]



selected_outcomes = ["X5YR1GPA", "X5YR1ATT", "X5YR1ERN", "X5REMRAT", "X5ENRATT"]

df_vars = df[[c for c in selected_vars if c in df.columns]].copy()
df_outcomes = df[[c for c in selected_outcomes if c in df.columns]].copy()

# Summary

print("Dropped all -5 columns:", len(all_neg5_cols))
print("Dropped loan columns:", len(loan_cols))
print("Dropped constant columns:", len(constant_cols))
print(f"Dropped high-missing columns (>={missing_threshold:.0%} missing):", len(high_missing_cols))
print("Wide cleaned shape:", df.shape)
print("Selected vars requested:", len(selected_vars))
print("Selected vars found:", df_vars.shape[1])
print("Model dataframe shape:", df_vars.shape)


display(df_vars.head())
display(df_vars.isna().mean().sort_values(ascending=False).head(20))


Raw shape: (23503, 9614)
Dropped all -5 columns: 932
Dropped loan columns: 21
Dropped constant columns: 1
Dropped high-missing columns (>=80% missing): 0
Wide cleaned shape: (23503, 8660)
Selected vars requested: 50
Selected vars found: 50
Model dataframe shape: (23503, 50)


,X1SEX,X1RACE,X1HISPANIC,X1WHITE,X1BLACK,X1PAREDU,X1FAMINCOME,X1SES,X1POVERTY130,X1HHNUMBER,...,X3TCREDSCI,X3TCREDENG,X3THIMATH,X3THISCI,X3T1CREDALG2,X3T1CREDCALC,X3T1CREDPHYS,X3T1CREDSTAT,X3TWHENALG1,X3HSCOMPSTAT
0,1,8,0,1,0,5,10,1.5644,0,3,...,3.0,4.0,9,1,1,0,0,1,4,1
1,2,8,0,1,0,3,3,-0.3699,0,6,...,3.5,4.5,6,2,0,0,0,0,5,1
2,2,3,0,0,1,7,6,1.2741,0,3,...,4.0,4.0,11,5,1,0,0,1,10,1
3,2,8,0,1,0,4,5,0.5498,0,2,...,3.5,4.0,6,1,1,0,1,0,4,1
4,1,8,0,1,0,4,9,0.1495,0,4,...,4.0,4.0,11,2,0,0,1,0,10,1


X1SEX           0.0
X3TCREDADV      0.0
X2TXMQUINT      0.0
X3TGPATOT       0.0
X3TGPAWGT       0.0
X3TGPAACAD      0.0
X3TGPASTEM      0.0
X3TNEWBASIC     0.0
X3TCREDTOT      0.0
X3TCREDACAD     0.0
X3TCREDSTEM     0.0
X3TCREDAPIB     0.0
X3TCREDHON      0.0
X1RACE          0.0
X3TCREDMAT      0.0
X3TCREDSCI      0.0
X3TCREDENG      0.0
X3THIMATH       0.0
X3THISCI        0.0
X3T1CREDALG2    0.0
dtype: float64

### Potential Predictor Columns
- Curriculum: 'X3TNEWBASIC', 

> Curriculum pertains to the intensity and composition of their coursework. Values indicate whether students completed a college-bound core curriculum, a college-bound core curriculum without computer science, a standard core curriculum without computer science, a minimal core curriculum, or other curricular patterns. This variable is indicative of coursework rigor and academic preparation prior to college.

- Grades (9-12, Academic Track, and Weighted): 'X3TGPA12TH', 'X3TGPA11TH', 'X3TGPA10TH', 'X3TGPA9TH', 'X3TGPAACAD','X3TGPAWGT', 

> Grades (gpa) pertains to the level of success a student was able to achieve throughout their high school career. Grade specific gpas represent annual academic performance on a standard gpa scale, allowing analysis of trends over time. The academic-track gpa (X3TGPAACAD) reflects performance specifically in college-preparatory courses, while the weighted GPA (X3TGPAWGT) adjusts for course difficulty by giving additional weight to honors, AP, or IB classes. Together, these GPA metrics provide a comprehensive measure of sustained academic performance and/or academic growth through high school.

- School Avg SAT Per School: 'C2AVGSATMATH', 'C2AVGSATREAD', 'C2AVGSATWRIT'

> These include the average SAT Math, Reading, and Writing scores for each student’s high school. These can be indicative of school competitiveness and peer academic ability. Higher average scores could suggest a more academically selective or high-performing school environment, which may influence students' college readiness through peer effects and instructional quality.

- School Avg ACT Per School: 'C2AVGACTCOMP', 'C2AVGACTMATH', 'C2AVGACTENG', 'C2AVGACTREAD', 'C2AVGACTSCI'

> The dataset also includes school average ACT performance. Similar to the school average SAT variables, these metrics capture the academic environment and competitiveness of each student's high school. Schools with higher average ACT scores might provide stronger academic preparation and academically engaged peer groups, which may indirectly affect students’ readiness for college coursework.

#### Column Information (Curriculum - 'X3TNEWBASIC')
- 1	College-bound core curriculum
- 2	Core curriculum
- 3	College-bound core without computer sciences
- 4	Core without computer sciences
- 5	Minimal core curriculum
- 6	All other patterns
- -8	Unit non-response
- TOTAL		23,503	99.99
- 1 UnW = Unweighted

### Potential Outcomes
- College GPA: "X5YR1GPA", "X5YR1ATT", "X5YR1ERN", "X5REMRAT", "X5ENRATT"

### Dataset #2
 as above, add any more copies of this that you need to given how many datasets you have

In [ ]:
## YOUR CODE TO LOAD/CLEAN/TIDY/WRANGLE THE DATA GOES HERE
## FEEL FREE TO ADD MULTIPLE CELLS PER SECTION 

## Results

### Exploratory Data Analysis

Instructions: replace the words in this subsection with whatever words you need to setup and preview the EDA you're going to do.   

Please explicitly load the fully wrangled data you will use from `data/02-processed`.  This is a good idea rather than forcing people to re-run the data getting / wrangling cells above.  Sometimes it takes a long time to get / wrangle data compared to reloading the fixed up dataset.

Carry out whatever EDA you need to for your project in the code cells below.  Because every project will be different we can't really give you much of a template at this point. But please make sure you describe the what and why in text here as well as providing interpretation of results and context.

Please note that you should consider the use of python modules in your work.  Any code which gets called repeatedly should be modularized. So if you run the same pre-processing, analysis or visualiazation on different subsets of the data, then you should turn that into a function or class.  Put that function or class in a .py file that lives in `modules/`.  Import the module you made and use it to get your work done.  For reference see `get_raw()` which is inside `modules/get_data.py`. 



#### Section 1 of EDA - please give it a better title than this

Some more words and stuff.  Remember notebooks work best if you interleave the code that generates a result with properly annotate figures and text that puts these results into context.

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [7]:
df_vars.columns

Index(['X1SEX', 'X1RACE', 'X1HISPANIC', 'X1WHITE', 'X1BLACK', 'X1PAREDU',
       'X1FAMINCOME', 'X1SES', 'X1POVERTY130', 'X1HHNUMBER', 'X2PAREDU',
       'X2FAMINCOME', 'X2SES', 'X2POVERTY130', 'X2HHNUMBER', 'X1CONTROL',
       'X1LOCALE', 'X1REGION', 'X1SCHOOLCLI', 'X2ENROLSTAT', 'X2ENRSTATSCH',
       'X2NUMHS', 'X2EVERDROP', 'X2DROPSTAT', 'X1TXMTSCOR', 'X1TXMQUINT',
       'X2TXMTSCOR', 'X2TXMQUINT', 'X3TGPATOT', 'X3TGPAWGT', 'X3TGPAACAD',
       'X3TGPASTEM', 'X3TNEWBASIC', 'X3TCREDTOT', 'X3TCREDACAD', 'X3TCREDSTEM',
       'X3TCREDAPIB', 'X3TCREDADV', 'X3TCREDHON', 'X3TCREDMAT', 'X3TCREDSCI',
       'X3TCREDENG', 'X3THIMATH', 'X3THISCI', 'X3T1CREDALG2', 'X3T1CREDCALC',
       'X3T1CREDPHYS', 'X3T1CREDSTAT', 'X3TWHENALG1', 'X3HSCOMPSTAT'],
      dtype='object')

In [ ]:
columns_remame = {
    'X3TNEWBASIC':'RIGOR',
    'X3TGPA12TH': 'GPA_12TH',
    'X3TGPA11TH': 'GPA_11TH',
    'X3TGPA10TH': 'GPA_10TH',
    'X3TGPA9TH': 'GPA_9TH',
    'X3TGPAACAD': 'GPA_ACAD',
    'X3TGPAWGT': 'GPA_WGT',
    'C2AVGSATMATH': 'SAT_MATH_AVG',
    'C2AVGSATREAD': 'SAT_READ_AVG',
    'C2AVGSATWRIT': 'SAT_WRIT_AVG',
}

#### Section 2 of EDA if you need it  - please give it a better title than this

Some more words and stuff.  Remember notebooks work best if you interleave the code that generates a result with properly annotate figures and text that puts these results into context.

In [ ]:
## YOUR CODE HERE
## FEEL FREE TO ADD MULTIPLE CELLS PER SECTION

## Ethics 

Instructions: Keep the contents of this cell. For each item on the checklist
-  put an X there if you've considered the item
-  IF THE ITEM IS RELEVANT place a short paragraph after the checklist item discussing the issue.
  
Items on this checklist are meant to provoke discussion among good-faith actors who take their ethical responsibilities seriously. Your teams will document these discussions and decisions for posterity using this section.  You don't have to solve these problems, you just have to acknowledge any potential harm no matter how unlikely.

Here is a [list of real world examples](https://deon.drivendata.org/examples/) for each item in the checklist that can refer to.

[![Deon badge](https://img.shields.io/badge/ethics%20checklist-deon-brightgreen.svg?style=popout-square)](http://deon.drivendata.org/)

### A. Data Collection
 - [X] **A.1 Informed consent**: If there are human subjects, have they given informed consent, where subjects affirmatively opt-in and have a clear understanding of the data uses to which they consent?
> All students are aware that their data would be used. No data was collected through force or secrecy.
 - [ ] **A.2 Collection bias**: Have we considered sources of bias that could be introduced during data collection and survey design and taken steps to mitigate those?
 
 - [X] **A.3 Limit PII exposure**: Have we considered ways to minimize exposure of personally identifiable information (PII) for example through anonymization or not collecting information that isn't relevant for analysis?
> No names, contact information, or other identifiers were collected or stored. Students will be differentiated using a unique number identifier, eliminating the possibility of PII.
 - [ ] **A.4 Downstream bias mitigation**: Have we considered ways to enable testing downstream results for biased outcomes (e.g., collecting data on protected group status like race or gender)?

### B. Data Storage
 - [X] **B.1 Data security**: Do we have a plan to protect and secure data (e.g., encryption at rest and in transit, access controls on internal users and third parties, access logs, and up-to-date software)?
> All data is from public sources and contains no PII; access is limited to project members.
 - [X] **B.2 Right to be forgotten**: Do we have a mechanism through which an individual can request their personal information be removed?
 > No, individuals will not be able to request for the removal of their data.
 - [X] **B.3 Data retention plan**: Is there a schedule or plan to delete the data after it is no longer needed?
 > Temporary copies of the dataset will be deleted from local machines once grading is complete. This minimizes long-term risk of unintended reuse. No archival use of the data is planned. Retention is strictly tied to the academic lifecycle of the project.

### C. Analysis
 - [ ] **C.1 Missing perspectives**: Have we sought to address blindspots in the analysis through engagement with relevant stakeholders (e.g., checking assumptions and discussing implications with affected communities and subject matter experts)?
 > We have no relevant stakeholders. 
 - [X] **C.2 Dataset bias**: Have we examined the data for possible sources of bias and taken steps to mitigate or address these biases (e.g., stereotype perpetuation, confirmation bias, imbalanced classes, or omitted confounding variables)?
 > These biases could influence correlations observed in the analysis. So, findings will be framed as descriptive rather than prescriptive. The analysis avoids claims about inherent ability or merit. Observed patterns are interpreted as outcomes shaped by broader educational contexts.
 - [X] **C.3 Honest representation**: Are our visualizations, summary statistics, and reports designed to honestly represent the underlying data?
> Axes, scales, and labels are chosen carefully to avoid misleading interpretations. Any uncertainty or variability in the data will be clearly shown when relevant. Visualizations will not be optimized to support a predetermined narrative. Alternative explanations for trends will be acknowledged.
 - [X] **C.4 Privacy in analysis**: Have we ensured that data with PII are not used or displayed unless necessary for the analysis?
 > No personally identifiable information will be used or displayed during analysis.
 - [X] **C.5 Auditability**: Is the process of generating the analysis well documented and reproducible if we discover issues in the future?
 > Code and analysis steps are version-controlled and clearly commented. This allows future reviewers to trace decisions and identify potential issues. Assumptions made during preprocessing aredocumented.

### D. Modeling
 - [X] **D.1 Proxy discrimination**: Have we ensured that the model does not rely on variables or proxies for variables that are unfairly discriminatory?
 > Features correlated with socioeconomic status or institutional advantage are not considered in the analysis, making it impossible for the model to rely on discriminatory variables.
 - [ ] **D.2 Fairness across groups**: Have we tested model results for fairness with respect to different affected groups (e.g., tested for disparate error rates)?
 > We have not yet tested any model results.
 - [X] **D.3 Metric selection**: Have we considered the effects of optimizing for our defined metrics and considered additional metrics?
 > Tradeoffs between accuracy, interpretability, and robustness are considered. Metrics are chosen to reflect the exploratory goals of the project. No single metric is treated as definitive. Performance results are interpreted within context rather than as absolute measures.
 - [X] **D.4 Explainability**: Can we explain in understandable terms a decision the model made in cases where a justification is needed?
 > Complex models are avoided where simpler explanations suffice. When predictions are discussed, their uncertainty is emphasized as to reduce the risk of inappropriate reliance on the model.
 - [] **D.5 Communicate limitations**: Have we communicated the shortcomings, limitations, and biases of the model to relevant stakeholders in ways that can be generally understood?

### E. Deployment
 - [ ] **E.1 Monitoring and evaluation**: Do we have a clear plan to monitor the model and its impacts after it is deployed (e.g., performance monitoring, regular audit of sample predictions, human review of high-stakes decisions, reviewing downstream impacts of errors or low-confidence decisions, testing for concept drift)?
 > We will not be deploying the model. For this reason, this ethical consideration is irrelevant.
 - [ ] **E.2 Redress**: Have we discussed with our organization a plan for response if users are harmed by the results (e.g., how does the data science team evaluate these cases and update analysis and models to prevent future harm)?
 > Since the model will not be deployed, this ethical issue does not apply.
 - [ ] **E.3 Roll back**: Is there a way to turn off or roll back the model in production if necessary?
 > As the model will not be used in practice, this ethical consideration can be disregarded.
 - [ ] **E.4 Unintended use**: Have we taken steps to identify and prevent unintended uses and abuse of the model and do we have a plan to monitor these once the model is deployed?
 > The model will not be implemented, so this ethical point is not pertinent.


## Team Expectations 

* *Everyone will contribute to their greatest extent on this project.*
* *Complete assigned task in a reasonable time frame (24-48 hours before next checkpoint).*
* *Communicate through iMessage for any concerns.*
* *Communicate newly added commits beforehand and gain approval from team members.*
* *Communicate obstructions or emergencies.*
* *Team members will address or assist anyone in any concern (or seek assistance from TA/Professor for complications)*

## Project Timeline Proposal


Specify your team's specific project timeline. Calendar starts 2/03 (initial project proposal) and ends 3/20 (final submission). Update times and task owners as needed.

| Meeting Date  | Meeting Time | Completed Before Meeting | Discuss at Meeting |
|---|---:|---|---|
| 2/03 | 5:00 PM | Initial project proposal draft; assign roles (All: Jaden, Trenton, Juhoan, Minh, Shawn) | Finalize research question, hypothesis, and timeline |
| 2/10 | 6:00 PM | Background research & locate candidate datasets (Trenton, Juhoan) | Review dataset options, data access steps, and ethics considerations |
| 2/17 | 6:00 PM | Download & store raw data into `data/00-raw/` (Jaden) | Review data schema and outline wrangling tasks; assign wrangling leads |
| 2/24 | 6:00 PM | Wrangling & initial EDA completed for a sample subset (Minh) | Review cleaned data, refine EDA, choose features for analysis |
| 3/03 | 6:00 PM | Complete wrangling & full EDA report draft (Shawn, Jaden) | Finalize analysis plan and modeling approach; split analysis tasks |
| 3/10 | 6:00 PM | Run initial analyses/models; produce preliminary results (Trenton, Minh) | Review model performance, iterate, and prepare figures for results |
| 3/17 | 6:00 PM | Draft results, figures, and discussion; begin final write-up (All) | Peer-review full project, address feedback, and finalize edits |
| 3/20 | Before 11:59 PM | Final project PDF, notebook, and code pushed to repo; group surveys submitted | Turn in Final Project & Group Project Surveys |